napari/utils/events/event.py

```diff
@@ -369,6 +369,20 @@ def source(self) -> Any:
     def source(self, s):
         self._source = None if s is None else weakref.ref(s)
 
+    def _is_core_napari_callback(self, callback: Union[CallbackRef, Callback]):
+        try:
+            if isinstance(callback, partial):
+                callback = callback.func
+            if not isinstance(callback, tuple):
+                return callback.__module__.startswith('napari.')
+            obj = callback[0]()
+            if obj is None:
+                return False
+            return obj.__module__.startswith('napari.')
+
+        except AttributeError:
+            return False
```

Comment:
I found this flow hard to follow, because I don't know all of the things that `callback` can be. Perhaps some comments would help? Like, what is `obj` expected to be?

I think I would also prefer a path that unifies the two non-False returns.

Reply:
Callback could be `Tuple[weakref, str[` representing a reference to python object and method name or function. So `obj` is derefered object which method callback should call. 

Did I should add documentation in this function, or next to the definition of `CallbackRef` and `Callback`?

Reply:
Comments describing this are fine. For me the guideline is that documentation should be local — I should be able to understand well enough a bit of code without having to jump around the codebase.

Reply:
@jni could you check docstring and comments? 

And I do not see good way to merge this two way of non False returns because these are two different objects. One is a function, the second one is an object. 

The only option that I see is to get the method from the object and assign it to the callback. but this introduces additional complexity for me.